.chloggen/configgrpc-metadata-interceptor.yaml

```diff
@@ -0,0 +1,26 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'enhancement'
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: configgrpc
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Apply configured Headers automatically
+
+# One or more tracking issues or pull requests related to the change
+issues: [12307]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  configgrpc now calls metadata.AppendToOutgoingContext automatically in an interceptor.
```

Comment:
Can we add here that exporters that do this manually will need to remove the manual code doing this?